app/questionnaire/questionnaire_schema.py

```diff
@@ -1,80 +1,50 @@
+import collections
+import itertools
 from app.validation.error_messages import error_messages
 
 
-class QuestionnaireSchema(object):
-
+class QuestionnaireSchema(object):  # pylint: disable=too-many-public-methods
     def __init__(self, questionnaire_json):
         self.json = questionnaire_json
-        self._groups_by_id = self._get_groups_by_id()
-        self._blocks_by_id = self._get_blocks_by_id()
-        self._answers_by_id = self._get_answers_by_id()
-        self.error_messages = self._get_error_messages()
-        self.aliases = self._get_aliases()
+        self._parse_schema()
 
-    def _get_groups_by_id(self):
-        groups = {}
-        for group in self.json.get('groups', []):
-            groups[group.get('id')] = group
+    @property
+    def sections(self):
+        return self._sections_by_id.values()
 
-        return groups
+    @property
+    def groups(self):
+        return self._groups_by_id.values()
 
-    def _get_blocks_by_id(self):
-        blocks = {}
-        for group in self.json.get('groups', []):
-            for block in group['blocks']:
-                blocks[block.get('id')] = block
+    @property
+    def blocks(self):
+        return self._blocks_by_id.values()
 
-        return blocks
+    @property
+    def questions(self):
+        return self._questions_by_id.values()
 
-    def _get_answers_by_id(self):
-        answers = {}
-        for block in self.get_blocks():
-            for question in self.get_questions_for_block(block):
-                for answer in question['answers']:
-                    answers[answer['id']] = answer
+    @property
+    def answers(self):
+        return self._answers_by_id.values()
```

Comment:
This and `def get_question(self, question_id):` are missing from our code coverage. Are they actually getting used anywhere? If so we need a test to confirm the output.